### DISCUSSION TITLE
Oversampling for imbalanced data

### DISCUSSION THREAD

**Discussion by YunchaoLiu** [Lightning-AI/pytorch-lightning#10005] [General]
Hi, I'm new to pytorch lightning. I'm trying to train on a very imbalanced dataset. I'm wondering if there is a way to do the oversampling with PL?

**Comment by rohitgr7** [comment]
PL doesn't take care of this. You can simply configure a weighted sampler and inject it to your datalaoder.
Ref: https://discuss.pytorch.org/t/how-to-handle-imbalanced-classes/11264/2

**Comment by daMichaelB** [comment]
Hey @rohitgr7 ,

i tried exactly that with creating a `ImageClassificationData` with the `from_data_frame` factory:

```python
ImageClassificationData.from_data_frame(
            ...
            sampler=WeightedRandomSampler
        )
```

It seems that the behaviour in `data_module._train_dataloader()` function has changed and the sampler is initialized there:

![image](https://user-images.githubusercontent.com/19696233/155719933-00ce97ee-7561-4dea-a523-dfefeebcce15.png)

I also found this in the docs:
<img width="1129" alt="image" src="https://user-images.githubusercontent.com/19696233/155720322-7e8ffe0b-ded0-42b0-aaec-945005686825.png">


It is expected to no longer pass a sampler instance, but a sampler class. I am wondering how i can pass all the parameters of the **WeightedRandomSampler** to this piece of code. I was not able to do it. 

Should i open up a ticket or do you have any idea ? Thank you

**Reply by ethanwharris** [reply]
Hey @daMichaelB Mind opening an issue on the Flash repo for this? We should support both sampler type (in the event that it needs a dataset argument) / partials and instantiated samplers (for samplers like weighted random which don't need it).

Thanks :smiley:

**Reply by daMichaelB** [reply]
Hello @ethanwharris. Sure thing! i will open up an issue. Thanks for the quick response
